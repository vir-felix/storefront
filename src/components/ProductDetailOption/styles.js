export default (theme) => ({
  variantOptionContainer: {
    display: "flex",
    width: `${theme.spacing.unit * 3.125}%`,
    maxWidth: `${theme.spacing.unit * 3.125}%`,
    padding: theme.spacing.unit * 1.25
  },
  optionButton: {
    "flex": "1 1 auto",
    "fontWeight": 600,
    "padding": theme.spacing.unit * 1.25,
    "borderRadius": "2px",
    "backgroundColor": theme.palette.primary.contrastText,
    "border": "1px solid",
    "borderColor": theme.palette.primary.borderColor,
    "&:hover": {
      backgroundColor: theme.palette.reaction.activeElementBackground
    }
  },
  optionText: {
    fontWeight: 500,
    fontSize: "14px"
  },
  isSelected: {
    backgroundColor: theme.palette.reaction.activeElementBackground
  }
});
