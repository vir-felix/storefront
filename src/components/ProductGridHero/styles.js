export default (theme) => ({
  heroImg: {
    width: "100%",
    height: "325px",
    objectFit: "cover"
  },
  productGridContainer: {
    maxWidth: theme.grid.productGridMaxWidth,
    marginLeft: "auto",
    marginRight: "auto"
  }
});
