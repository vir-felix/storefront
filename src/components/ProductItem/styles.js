export const styles = (theme) => ({
  "@keyframes spin": {
    from: { transform: "rotate(0deg)" },
    to: { transform: "rotate(-360deg)" }
  },
  "root": {},
  "productInfo": {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    paddingTop: 10
  },
  "productMedia": {
    backgroundColor: theme.palette.primary.contrastText,
    position: "relative"
  },
  "img": {
    height: "auto",
    width: "100%"
  },
  "imgLoading": {
    alignItems: "center",
    backgroundColor: "transparent",
    display: "flex",
    fontSize: 48,
    height: 100,
    justifyContent: "center",
    left: "calc(50% - 50px)",
    position: "absolute",
    top: "calc(50% - 50px)",
    width: 100
  },
  "loadingIcon": {
    color: theme.palette.primary.light,
    fontSize: "inherit",
    animationName: "spin",
    animationDuration: theme.transitions.duration.standard * 2,
    animationTimingFunction: theme.transitions.easing.sharp,
    animationIterationCount: "infinite",
    animationFillMode: "both"
  },
  "link": {
    ...theme.typography.body2,
    "textAlign": "left",
    "&:hover": {
      color: theme.palette.secondary.main
    }
  }
});
