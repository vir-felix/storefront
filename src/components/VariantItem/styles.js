export default (theme) => ({
  variantButton: {
    "display": "flex",
    "justifyContent": "space-between",
    "textTransform": "none",
    "width": "100%",
    "borderRadius": "2px",
    "backgroundColor": theme.palette.primary.contrastText,
    "border": "1px solid",
    "borderColor": theme.palette.primary.contrastText,
    "paddingTop": theme.spacing.unit * 1.25,
    "paddingBottom": theme.spacing.unit * 1.25,
    "paddingLeft": theme.spacing.unit * 2.5,
    "paddingRight": theme.spacing.unit * 2.5,
    "&:hover": {
      border: "1px solid",
      borderColor: theme.palette.reaction.activeElementBorderColor,
      backgroundColor: theme.palette.primary.contrastText
    }
  },
  activeVariant: {
    border: "1px solid",
    borderColor: theme.palette.reaction.activeElementBorderColor
  }
});
