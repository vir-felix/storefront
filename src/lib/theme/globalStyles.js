export default
<style global jsx>
  {`
    a {
      color: inherit;
      text-decoration: none;
    }
  `}
</style>;
