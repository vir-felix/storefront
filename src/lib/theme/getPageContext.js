import { SheetsRegistry } from "jss";
import { createGenerateClassName } from "@material-ui/core/styles";
import theme from "./reactionTheme";

/**
 * @summary Creates and returns a page context object
 * @returns {Object} Page context object
 */
function createPageContext() {
  return {
    theme,
    // This is needed in order to deduplicate the injection of CSS in the page.
    sheetsManager: new Map(),
    // This is needed in order to inject the critical CSS.
    sheetsRegistry: new SheetsRegistry(),
    // The standard class name generator.
    generateClassName: createGenerateClassName()
  };
}

/**
 * Either creates a new or returns an existing page context object. If called from Node, creates
 * a new object every time. Otherwise creates and caches an object in the browser and
 * subsequent calls will return that.
 *
 * @summary Gets a page context object
 * @returns {Object} Page context object
 */
export default function getPageContext() {
  // Make sure to create a new context for every server-side request so that data
  // isn't shared between connections (which would be bad).
  if (!process.browser) {
    return createPageContext();
  }

  // Reuse context on the client-side.
  if (!global.__INIT_MATERIAL_UI__) {
    global.__INIT_MATERIAL_UI__ = createPageContext();
  }

  return global.__INIT_MATERIAL_UI__;
}
