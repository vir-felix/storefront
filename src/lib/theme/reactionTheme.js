import { createMuiTheme } from "material-ui/styles";

const theme = createMuiTheme({
  palette: {
    primary: {
      light: "#26B0F9",
      main: "#1999DD",
      dark: "#172F3C",
      contrastText: "#FFFFFF",
      borderColor: "#CCCCCC"
    },
    secondary: {
      light: "#5d8ea9",
      main: "#5E7480",
      dark: "#1D1D1D",
      contrastText: "#000000"
    },
    error: {
      light: "#E54F5D",
      main: "#CD3F4C",
      dark: "#3C1F21",
      contrastText: "#FFFFFF"
    },
    reaction: {
      activeElementBorderColor: "#94E8D1",
      activeElementBackground: "#E6E6E6"
    }

  },
  typography: {
    fontFamily: "Source Sans Pro, Helvetica Neue, Helvetica, sans-serif",
    fontSize: 16,
    fontWeightLight: 200,
    fontWeightRegular: 400,
    fontWeightMedium: 700,
    fontWeightBold: 700
  }
});

export default theme;
