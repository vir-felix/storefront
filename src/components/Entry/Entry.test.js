import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { MuiThemeProvider } from "@material-ui/core/styles";
import theme from "lib/theme/reactionTheme";
import Entry from "./";

test("basic snapshot", () => {
  const component = renderer.create(shallow(<MuiThemeProvider theme={theme}>
    <Entry />
  </MuiThemeProvider>).get(3));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
