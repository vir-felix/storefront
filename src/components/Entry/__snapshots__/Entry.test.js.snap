// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<div
  className="MuiGrid-container-4"
>
  <div
    className="MuiGrid-item-5 MuiGrid-grid-xs-12-44 MuiGrid-grid-md-7-67"
  >
    <div
      className="SkEntry-loginWrapper-1"
    >
      <h2
        className="MuiTypography-root-101 MuiTypography-title-107 MuiTypography-gutterBottom-119"
      >
        Returning Customer
      </h2>
      <div
        className="SkEntry-loginButton-2 Button__ButtonDiv-bnfrK eloVkm"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
        title={undefined}
      >
        <div>
          Login
        </div>
        <div
          className="Button__SpinnerWrap-bgOcuM cCKgPo"
          style={
            Object {
              "opacity": 0,
              "paddingLeft": 0,
              "width": 0,
            }
          }
        >
          <svg
            className="spinner__SpinnerSvg-iBTUvT bunzNs"
            height="40px"
            style={
              Object {
                "enableBackground": "new 0 0 50 50",
              }
            }
            version="1.1"
            viewBox="0 0 50 50"
            width="40px"
            x="0px"
            y="0px"
          >
            <path
              d="M43.935,25.145c0-10.318-8.364-18.683-18.683-18.683c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615c8.072,0,14.615,6.543,14.615,14.615H43.935z"
              fill="#000"
            >
              <animateTransform
                attributeName="transform"
                attributeType="xml"
                dur="1s"
                from="0 25 25"
                repeatCount="indefinite"
                to="360 25 25"
                type="rotate"
              />
            </path>
          </svg>
        </div>
      </div>
      <div
        className="SkEntry-loginButton-2 Button__ButtonDiv-bnfrK jNrbdB"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
        title={undefined}
      >
        <div>
          Create a new account
        </div>
        <div
          className="Button__SpinnerWrap-bgOcuM fcVXSk"
          style={
            Object {
              "opacity": 0,
              "paddingLeft": 0,
              "width": 0,
            }
          }
        >
          <svg
            className="spinner__SpinnerSvg-iBTUvT bunzNs"
            height="40px"
            style={
              Object {
                "enableBackground": "new 0 0 50 50",
              }
            }
            version="1.1"
            viewBox="0 0 50 50"
            width="40px"
            x="0px"
            y="0px"
          >
            <path
              d="M43.935,25.145c0-10.318-8.364-18.683-18.683-18.683c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615c8.072,0,14.615,6.543,14.615,14.615H43.935z"
              fill="#000"
            >
              <animateTransform
                attributeName="transform"
                attributeType="xml"
                dur="1s"
                from="0 25 25"
                repeatCount="indefinite"
                to="360 25 25"
                type="rotate"
              />
            </path>
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-item-5 MuiGrid-grid-xs-12-44 MuiGrid-grid-md-5-65"
  >
    <div
      className="SkEntry-guestWrapper-3"
    >
      <h2
        className="MuiTypography-root-101 MuiTypography-title-107 MuiTypography-gutterBottom-119"
      >
        Guest Checkout
      </h2>
      <div
        className={null}
        style={Object {}}
      >
        <div
          className="required Field__StyledField-kQcQjL lnhSLv"
        >
          <label
            className="Field__StyledLabel-gsQloH iDRsBJ"
            htmlFor={undefined}
          >
            Email Address
          </label>
          <div
            className="TextInput__InputWrapper-dGJJJl bGPXhB"
            value=""
          >
            <input
              className="TextInput__StyledInput-loReSQ cKYvwP"
              id="email_GuestForm_1"
              maxLength={undefined}
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholder="Email address"
              readOnly={false}
              type="email"
              value=""
            />
          </div>
          <span
            className="Field__StyledHelpText-iiSdPK dTOPZc"
          >
            You will have the option to create an account and save your details after checkout.
          </span>
        </div>
        <div
          className="GuestForm__FormAction-fhrWsN gMhMif"
        >
          <div
            className="Button__ButtonDiv-bnfrK hRmBpg"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
            title={undefined}
          >
            <div>
              Continue as guest
            </div>
            <div
              className="Button__SpinnerWrap-bgOcuM fcVXSk"
              style={
                Object {
                  "opacity": 0,
                  "paddingLeft": 0,
                  "width": 0,
                }
              }
            >
              <svg
                className="spinner__SpinnerSvg-iBTUvT bunzNs"
                height="40px"
                style={
                  Object {
                    "enableBackground": "new 0 0 50 50",
                  }
                }
                version="1.1"
                viewBox="0 0 50 50"
                width="40px"
                x="0px"
                y="0px"
              >
                <path
                  d="M43.935,25.145c0-10.318-8.364-18.683-18.683-18.683c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615c8.072,0,14.615,6.543,14.615,14.615H43.935z"
                  fill="#000"
                >
                  <animateTransform
                    attributeName="transform"
                    attributeType="xml"
                    dur="1s"
                    from="0 25 25"
                    repeatCount="indefinite"
                    to="360 25 25"
                    type="rotate"
                  />
                </path>
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
